app/code/community/Paylike/Paymentmethod/Model/Paymentmethod.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+require_once dirname(__FILE__) . '/../Model/Paylike.php';
+class Paylike_Paymentmethod_Model_Paymentmethod extends Mage_Payment_Model_Method_Cc
+{
+    const REQUEST_TYPE_AUTH_CAPTURE = 'AUTH_CAPTURE';
+    const REQUEST_TYPE_AUTH_ONLY = 'AUTH_ONLY';
+
+    protected $_isGateway = true;
+    protected $_canAuthorize = true;
+    protected $_canUseCheckout = true;
+    protected $_canCapture = true;
+    protected $_canCapturePartial = false;
+    protected $_canRefund = true;
+    protected $_canRefundInvoicePartial = true;
+    protected $_canVoid = true;
+    protected $_code = 'paymentmethod';
+    protected $_formBlockType = 'paymentmethod/form_paymentmethod';
+    protected $_infoBlockType = 'paymentmethod/info_paymentmethod';
+
+    public function authorize(Varien_Object $payment, $amount)
+    {
+        $payment->setTransactionId($payment->getPaylikeTransactionId());
+        $payment->setIsTransactionClosed(0);
+        return $this;
+
+    }
+
+    public function capture(Varien_Object $payment, $amount)
+    {
+        $arr = array('amount' => $amount * 100,
+            'currency' => Mage::app()->getStore()->getCurrentCurrencyCode(),
+            'descriptor' => 'Capture',
+        );
+        if ($amount <= 0) {
+            $errormsg= $this->__('Invalid amount for authorization.');
+            Mage::throwException($errormsg);
+        }
+        $paylike = new Paylike($this->getApiKey());
+        if (!$payment->getLastTransId()) {
+            $payment->setLastTransId($payment->getPaylikeTransactionId());
+        }
+
+        $result = $paylike->transactions->capture($payment->getLastTransId(), $arr);
+        if ($result == false) {
+            $errormsg = $this->__('Transaction failed');
+            Mage::throwException($errormsg);
+        } else {
+
+            $payment->setTransactionId($payment->getPaylikeTransactionId());
+            $payment->setIsTransactionClosed(1);
+        }
+        return $this;
+    }
+
+    public function refund(Varien_Object $payment, $amount)
+    {
+        $arr = array('amount' => $amount * 100,
+            'descriptor' => 'refund',
```

Comment:
This is the text that will appear on the customer's credit card statement. It should help the customer remember what the transaction is about. I do not think the text "refund" will be that helpful.
It is better to remove it as we will then fall back to the descriptor of the transaction.